allennlp/training/trainer.py

```diff
@@ -1442,3 +1061,12 @@ def from_partial_objects(
             num_gradient_accumulation_steps=num_gradient_accumulation_steps,
             use_amp=use_amp,
         )
+
+
+DEFAULT_CALLBACKS = (
+    Lazy(SanityChecksCallback),
+    Lazy(ConsoleLoggerCallback),
+)
+"""
+The default callbacks used by `GradientDescentTrainer`.
+"""
```

Comment:
@AkshitaB, I think I mentioned this idea to you in our 1-1 two weeks ago? Instead of using flags to disable certain default callbacks (like `run_sanity_checks: bool`), we could just specify the default callbacks like this. Let me know what you think. I'm not 100% sure this is better.

Reply:
You could just set `trainer.callbacks: []` in the config.

Reply:
I don't like the flags either, but I think it's better than setting the callbacks. Otherwise you specify the wandb callback, and unbekownst to you, that disables the sanity checks. That's not terribly intuitive.

Reply:
Very good point. How about we have just one flag called `disable_default_callbacks: bool = False`. When that's false, we always add `DEFAULT_CALLBACKS` to the callbacks (being careful not to duplicate them).

Reply:
Sure, though in general I prefer flags to be positive, i.e., `enable_default_callbacks: bool = True`. It gets confusing when you write code like `disable_something = !skip_custom_ablation_code` or something like that.

Reply:
Aren't you skiing or something? :-P

Reply:
Just got back 😜

Okay, I'll do that. I like the positivity as well!

Reply:
Done